tipg/db.py

```diff
@@ -29,38 +29,48 @@
     custom_sql = list(pathlib.Path(user_sql_dir).glob("*.sql"))
 
 
-async def con_init(conn):
-    """Use orjson encoder/decoder and register custom SQL functions."""
-    await conn.set_type_codec(
-        "json", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
-    )
-    await conn.set_type_codec(
-        "jsonb", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
-    )
-
-    await conn.execute(
-        """
+class con_init:
+    """Connection creation."""
+
+    schemas: List[str]
+
+    def __init__(self, schemas: Optional[List[str]] = None) -> None:
+        """Init."""
+        self.schemas = schemas or []
+
+    async def __call__(self, conn: asyncpg.Connection):
+        """Create connection."""
+        await conn.set_type_codec(
+            "json", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
+        )
+        await conn.set_type_codec(
+            "jsonb", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
+        )
+        schemas = ",".join(["pg_temp"] + self.schemas)
```

Comment:
The order of the path seems important.

cc @bitner

Reply:
We want to make sure that pg_temp is the first so that any new functions/views/tables created when we load sql from files are created into the pg_temp schema - if you don't specify a schema, new objects are written into the first schema in the search_path. We should also make sure to add the function_schemas from the environment as well.

Reply:
Ahh never mind that last part of the comment, I see now below that self.schemas is already combined.

Reply:
In other projects (pygeoapi) it's needed to be `search_path: [published, public]` if you used `search_path: [public, published]` it wouldn't pick up `published`

Reply:
Oo so I found why I can't make `reproductible` test, because I use `Set` https://github.com/developmentseed/tipg/pull/48/files#diff-772cfc11cca1ff02c30e13c6c3aa3ae382b238719d6eaaa11fa1df9c864e0a1fR163 then the order of schemas changes 😬

Reply:
fixed by unifying the schemas options into one (for both tables and functions)